connect/api/src/main/java/org/apache/kafka/connect/data/SchemaBuilder.java

```diff
@@ -181,7 +181,7 @@ public SchemaBuilder doc(String doc) {
 
     @Override
     public Map<String, String> parameters() {
-        return Collections.unmodifiableMap(parameters);
+        return parameters == null ? null: Collections.unmodifiableMap(parameters);
```

Comment:
If that's what the interface specifies, then the change looks good independently of whether a NPE is thrown  or not (the severity of the issue does depend on that aspect though).